cve_bin_tool/checkers/libevent.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+
+CVE checker for Libevent
+
+https://www.cvedetails.com/vulnerability-list/vendor_id-15590/product_id-32303/Libevent-Project-Libevent.html
+
+"""
+
+from cve_bin_tool.checkers import Checker
+
+
+class LibeventChecker(Checker):
+    CONTAINS_PATTERNS = [
+        r"Added a signal to event base %p with signals already added to event_base %p.  Only one can have signals at a time with the %s backend.  The base with the most recently added signal or the most recent event_base_loop() call gets preference; do not rely on this behavior in future Libevent versions.",
+        r"Error from libevent when adding event"
+        r"Error from libevent when adding event for DNS server",
+        r"Error from libevent when adding timer event",
+    ]
+
+    FILENAME_PATTERNS = [
+        r"libevent.so",
+    ]
+
+    VERSION_PATTERNS = [
+        r"([0-9]\.[0-9]\.[0-9]{1,2})-stable",
```

Comment:
This pattern is no good. There's not enough here to ensure that you found libevent and not something else that includes -stable in their data or filenames.  I'd be very concerned about future false positives with this one, to the point where I'd rather not add it at all if this is the best pattern available.

Can you take a look at some multi-line options and see if you can find something which includes something like "libevent" (or possibly "event") or a string long enough to be unique to libevent.  We need to reduce the false positive risk and tie the detection specifically to libevent, and if that's not possible we should close both this and the checker request with a note that no viable signature could be found.

Reply:
For this I've added multi-line version patterns; Kindly check this please.